### PR TITLE
Lock 'Choose navigation' page when the step by step is scheduled 

### PR DESCRIPTION
**Pull request by AgaDufrat**
## What
When a user has scheduled a step by step for publishing, they should not be able to make any further changes to that step by step. 
- User _cannot_ change the navigation setting
- User _can_ view the navigation setting

**Not scheduled**
not_scheduled

**Scheduled for publishing**
locked

## Why
This prevents any unexpected edits being published in error or too early.

[Trello card](https://trello.com/c/0yrygTBl/88-lock-a-step-by-step-that-has-been-scheduled-for-publication-on-the-choose-navigation-pages)

### CODE REVIEW

File: spec/models/navigation_rule_spec.rb [hunk 1]
```diff
@@ -139,4 +139,91 @@
...
+          {
+            text: "Never show navigation",
+            value: "never",
+            selected: false
+          }
+        ]
+        expect(resource.options_with_selected).to eq(expected_options)
+      end
+    end
+    context "include_in_links is set to conditionally" do
```
Comment (leenagupte) [inline spec/models/navigation_rule_spec.rb]: Minor: Can you add a line break here?

File: spec/factories.rb [hunk 2]
```diff
@@ -54,6 +54,14 @@
     end
   end
 
+  factory :step_by_step_page_with_secondary_content_and_navigation_rules, parent: :step_by_step_page_with_steps do
```
Comment (leenagupte) [inline spec/factories.rb]: I'm not why this factory has been added? Could the existing `step_by_step_page_with_navigation_rules` be used instead?

File: spec/models/navigation_rule_spec.rb [hunk 3]
```diff
@@ -239,7 +239,7 @@
       )
       expect(resource.selected_option_text).to eq("Always show navigation")
     end
-
+
```
Comment (leenagupte) [inline spec/models/navigation_rule_spec.rb]: Can you fix this up?

File: spec/factories.rb (hunk 2 above)
Comment (AgaDufrat) [reply spec/factories.rb]: `step_by_step_page_with_navigation_rules` doesn't have secondary content links, which we also need in this test

**Pull request closed without merging**